File: problem/bool-not/6/main.py

# main.py: main file for solutions to the "bool-not" exercise
#
# Copyright (C) 2020  Rudy Matela
# All rights reserved.
#
#
# This file is part of the CSCX Online Judge.
#
# This is to be appended to the submitted file.

# standard input is already processed
# as the submitted programmed is run when imported

with open("in.txt") as filein:
	for line in filein:
		p = line.strip() == "true"
		p = negation(p)
		print("true" if p else "false")

File: problem/bool-not/6/main.lua
-- main.lua: main file for solution to the bool-not problem
--
-- Copyright (C) 2020-2021  Rudy Matela
-- All rights reserved.
--
--
-- This file is part of the CScx Online Judge.
--
-- This is appended to the submitted Lua program and tests the negation function.
--
-- The submitted file processes the standard input and this processes the
-- "in.txt" file.

io.input("in.txt")
p = io.read("*line")
while p do
  p = p == "true"
  print(negation(p))
  p = io.read("*line")
end
